kanbananza-backend/src/main/services/card.js

```diff
@@ -0,0 +1,40 @@
+import Card from "../models/card";
+import HttpError from "../http_error";
+
+const createCard = async ({ name, columnId, order }) => {
+  if (await Card.exists({ columnId, name })) {
+    throw new HttpError({
+      code: 400,
+      message: `Card name already in use for column with id ${columnId}.`,
+    });
+  }
+  try {
+    return await Card.create({ name, columnId, order });
+  } catch (e) {
+    if (e instanceof Error.ValidationError) {
+      throw new HttpError({
+        code: 400,
+        message: "Invalid field(s)",
+        body: Object.values(e.errors).map((error) => error.message),
+      });
```

Comment:
this should throw ValidationErrors that are caught and attached to HttpError in controller

Reply:
you can check the other services